go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -125,27 +125,22 @@ func (oc *Controller) updateEgressFirewall(oldEgressFirewall, newEgressFirewall
 func (oc *Controller) deleteEgressFirewall(egressFirewall *egressfirewallapi.EgressFirewall) []error {
 	klog.Infof("Deleting egress Firewall %s in namespace %s", egressFirewall.Name, egressFirewall.Namespace)
 
-	nsInfo, err := oc.waitForNamespaceLocked(egressFirewall.Namespace)
-	if err != nil {
-		return []error{fmt.Errorf("failed to wait for namespace %s event (%v)",
-			egressFirewall.Namespace, err)}
+	nsInfo := oc.getNamespaceLocked(egressFirewall.Namespace)
+	if nsInfo != nil {
```

Comment:
if nsInfo == nil why we don't return, we don't need to delete anything since the namespace was already deleted,no?
```suggestion
	if nsInfo == nil {
	    return nil
	 }
	 nsInfo.egressFirewallPolicy = nil
	nsInfo.Unlock() 
	....
```

Reply:
if is not the case we should add a comment explaining this logic, it's confusing delete things from a namespace that does not exist :)

Reply:
there's no handler for deleteEgressFirewall in namespace watcher. So I don't think we can rely on the namespace handler to delete the egress firewall.

Reply:
does it make sense to add that though? when a namespace delete happens, all contained objects should get deleted?

Reply:
@JacobTanenbaum wdyt?

Reply:
I don't think the namespace should delete the egressfirewall. When a namespace is deleted it is my understanding that all the objects inside it get deleted and we don't as far as I can see explicitly delete all pods in a namespace from the namespace handler when a namespace is deleted.

Reply:
When the namespace is deleted it will cause pod delete events. However in this case, if a namespace is deleted will it signal an egressFirewall delete event? I suppose it should, since it is namespace scoped (but someone should check). Either way this code will protect us from getting the events out of order.

Reply:
Yes, when a namespace is deleted, every namespaced object in that namespace gets deleted as well. But there's no guarantee what order we'll process the events in, so the nsInfo may or may not exist at this point.